src/Isotope.cpp

```diff
@@ -0,0 +1,607 @@
+#include "Isotope.h"
+
+int Isotope::_n = 0;
+
+
+/**
+ * @brief Constructor sets the ID, unique ID, and name for the Isotope.
+ * @param id the user-specified optional Isotope ID
+ * @param name the user-specified name for the Isotope
+ */
+Isotope::Isotope(int id, const char* name) {
+  _uid = _n;
+  _id = id;
+  _n++;
+  
+  _name = NULL;
+  setName(name);
+  
+  _sigma_t = NULL;
+  _sigma_a = NULL;
+  _sigma_s = NULL;
+  _sigma_f = NULL;
+  _nu_sigma_f = NULL;
+  _chi = NULL;
+  
+  return;
+}
+
+
+/**
+ * @brief Destructor deletes all cross-section data structures from memory.
+ */
+Isotope::~Isotope() {
+  if (_sigma_t != NULL)
+    delete [] _sigma_t;
+
+  if (_sigma_a != NULL)
+    delete [] _sigma_a;
+    
+  if (_sigma_s != NULL)
+    delete [] _sigma_s;
+
+  if (_sigma_f != NULL)
+    delete [] _sigma_f;
+
+  if (_nu_sigma_f != NULL)
+    delete [] _nu_sigma_f;
+
+  if (_chi != NULL)
+    delete [] _chi;
+  
+}
+
+
+
+/**
+ * @brief Return the isotope's unique ID.
+ * @return the isotope's unique ID
+ */
+int Isotope::getUid() const {
+  return _uid;
+}
+
+
+/**
+ * @brief Return the isotope's user-defined ID
+ * @return the isotope's user-defined ID
+ */
+int Isotope::getId() const {
+  return _id;
+}
+
+
+/**
+ * @brief Return the user-defined name of the Isotope
+ * @return the Isotope name
+ */
+char* Isotope::getName() const {
+  return _name;
+}
+
+
+/**
+ * @brief Sets the name of the Isotope
+ * @param name the Isotope name string
+ */
+void Isotope::setName(const char* name) {
+  int length = strlen(name);
+
+  if (_name != NULL)
+    delete [] _name;
+
+  /* Initialize a character array for the Isotope's name */
+  _name = new char[length+1];
+
+  /* Copy the input character array Isotope name to the class attribute name */
+  for (int i=0; i <= length; i++)
+    _name[i] = name[i];
+}
+
+
+/**
+ * @brief Returns the number of energy groups for this isotope's nuclear data.
+ * @return the number of energy groups
+ */
+int Isotope::getNumEnergyGroups() const {
+  return _num_groups;
+}
+
+
+/**
+ * @brief Return the array of the isotope's total cross-sections.
+ * @return the pointer to the isotope's array of total cross-sections
+ */
+FP_PRECISION* Isotope::getSigmaT() {
```

Comment:
Are the ``Isotope's`` cross-sections micros or macros? Based on what I saw in your new sample input files, I assume micros. You might want to add a ``@details`` doxygen comment block to these setter methods to explicitly specify what the routine expects.